tests/unit/report_writers/test_sarif_report_writer.py

```diff
@@ -0,0 +1,563 @@
+        self, mock_file, mock_makedirs
+    ):
+        """Test that corrupted SARIF input raises appropriate exception."""
+        with self.assertRaises(json.JSONDecodeError):
+            generate_sarif_report_with_ai_analysis(self.sample_data, self.mock_config)
+
+
+class TestInjectAnalysisResults(SarifTestBase):
+    """Test cases for the _inject_analysis_results function."""
+
+    def setUp(self):
+        """Set up test fixtures."""
+        self.mock_config = self.create_mock_config()
+        self.sarif_data = self.create_sample_sarif_data()
+
+        # Create sample analysis data
+        self.sample_issues = TestUtils.create_sample_issues(count=2)
+        self.analysis_data = []
+        for i, issue in enumerate(self.sample_issues):
+            summary_info = Mock(spec=SummaryInfo)
+            summary_info.llm_response = Mock(spec=AnalysisResponse)
+            summary_info.llm_response.investigation_result = (
+                CVEValidationStatus.FALSE_POSITIVE.value
+                if i == 0
+                else CVEValidationStatus.TRUE_POSITIVE.value
+            )
+            summary_info.llm_response.short_justifications = f"Test justification {i+1}"
+            summary_info.llm_response.recommendations = [f"Test recommendation {i+1}"]
+            summary_info.llm_response.is_true_positive.return_value = (
+                i == 1
+            )  # Second issue is true positive
+            self.analysis_data.append((issue, summary_info))
+
+    @patch("report_writers.sarif_report_writer._update_tool_info")
+    @patch("report_writers.sarif_report_writer._add_suppression")
+    def test_given_analysis_data_when_injecting_then_updates_tool_and_adds_suppressions(
+        self, mock_add_suppression, mock_update_tool
+    ):
+        """Test that analysis injection updates tool info and adds suppressions to all results."""
+        # Execute
+        result = _inject_analysis_results(self.sarif_data, self.analysis_data, self.mock_config)
+
+        # Verify tool info was updated
+        mock_update_tool.assert_called_once_with(self.sarif_data, self.mock_config)
+
+        # Verify suppressions were added for each result
+        self.assertEqual(mock_add_suppression.call_count, 2)
+
+        # Verify the calls were made with correct parameters
+        for i in range(2):
+            call_args = mock_add_suppression.call_args_list[i]
+            self.assertEqual(
+                call_args[0][0], self.sarif_data["runs"][0]["results"][i]
+            )  # SARIF result
+            self.assertEqual(call_args[0][1], self.analysis_data[i][0])  # Issue
+            self.assertEqual(call_args[0][2], self.analysis_data[i][1])  # SummaryInfo
+
+        # Verify the original SARIF data is returned (modified in place)
+        self.assertEqual(result, self.sarif_data)
+
+    def test_given_empty_analysis_data_with_sarif_results_when_injecting_then_raises_error(self):
```

Comment:
In this case, we should still upload the original SARIF file to the given storage bucket.